src/sparsebase/io/mtx_writer.cc

```diff
@@ -0,0 +1,320 @@
+        }
+
+        //Symmetry check
+        bool saidSymmetric = (symmetry_ == "symmetric" || symmetry_ == "skew-symmetric" || symmetry_ == "hermitian");
+        int NNZ = coo->get_num_nnz();
+        int count_symmetric = 0;
+        int count_diagonal = 0;
+        if (saidSymmetric && dimensions[0] == dimensions[1]) {
+            for (int i = 0; i < coo->get_num_nnz(); ++i) {
+                if (row[i] != col[i]) { // Non-diagonal entry, check symmetric counterpart
+                    bool found_symmetric = false;
+                    for (int j = 0; j < coo->get_num_nnz(); ++j) {
+                      if (symmetry_ == "skew-symmetric")
+                      {
+                        if (row[j] == col[i] && col[j] == row[i] && val[j] == -val[i]) {
+                            found_symmetric = true;
+                            count_symmetric++;
+                            break;
+                        }
+                      }
+                      else
+                      {
+                        if (row[j] == col[i] && col[j] == row[i] && val[j] == val[i]) {
+                            found_symmetric = true;
+                            count_symmetric++;
+                            break;
+                        }
+                      }
+                        
+                    }
+                    if (!found_symmetric) {
+                        throw utils::ReaderException("Matrix is not symmetric!");
+                    }
+                }
+                else //diagonal
+                {
+                  count_diagonal++;
+                }
+            }
+            if(symmetry_ == "skew-symmetric")
+            {
+              NNZ -= (count_symmetric/2) + count_diagonal;
+            }
+            else
+            {
+              NNZ -= (count_symmetric/2);
+            }
+            
+        }
+      
+        //dimensions and nnz
+        if (format_ == "array")
+        {
+          mtxFile << dimensions[0] << " " << dimensions[1] << "\n";
+        }
+        else //cordinate
+        {
+          mtxFile << dimensions[0] << " " << dimensions[1] << " " << NNZ << "\n";
+        }
+
+        //write data lines
+        if (format_ == "array")
+        {
+
+          //sort according to column values
+          std::vector<std::tuple<IDType, IDType, ValueType>> sort_vec;
+          for (int i = 0; i < coo->get_num_nnz(); i++) {
+            sort_vec.emplace_back(col[i], row[i], val[i]);
+          }
+          std::sort(sort_vec.begin(), sort_vec.end(),
+                    [](std::tuple<IDType, IDType, ValueType> t1, std::tuple<IDType, IDType, ValueType> t2) {
+                      if (std::get<0>(t1) == std::get<0>(t2)) {
+                        return std::get<1>(t1) < std::get<1>(t2);
+                      }
+                      return std::get<0>(t1) < std::get<0>(t2);
+                    });
+
+          for (int i = 0; i < coo->get_num_nnz(); i++) {
```

Comment:
This loop doesn't seem to be doing anything, I guess it was missed :)

Reply:
deleted :)